components/YarnItem/index.js

```diff
@@ -0,0 +1,114 @@
+import styled from "styled-components";
+import StyledInput from "../StyledInput";
+import { useEffect } from "react";
+// import { useEffect } from "react";
+export default function YarnItem({
+  defaultValue,
+  isEdit,
+  setYarnData,
+  yarnCount,
+  yarnData,
+}) {
+  useEffect(() => {
+    if (isEdit) {
+      setYarnData(defaultValue.yarn);
+    }
+  }, []);
+  // useEffect(() => {
+  //   if (isEdit) {
+  //     setYarnData(defaultValue.yarn);
+  //   }
+  // }, [yarnData]);
+
+  function handleInputChange(event) {
+    const { name, value } = event.target;
+    setYarnData((prevData) => {
+      const updatedData = [...prevData];
+      updatedData[yarnCount] = {
+        ...updatedData[yarnCount],
+        [name]: value,
+      };
+      console.log(yarnData);
```

Comment:
I am not sure if this console.log is allowed to stay

Reply:
Thanks Rüveyda, i delete these code. They are unnecessary 😄